internal/configs/sharding/allowlist.go

```diff
@@ -0,0 +1,47 @@
+package shardingconfig
+
+import (
+	"fmt"
+
+	bls_cosi "github.com/harmony-one/harmony/crypto/bls"
+)
+
+type Allowlist struct {
+	MaxLimitPerShard int
+	BLSPublicKeys    []bls_cosi.PublicKeyWrapper
+}
+
+func BLS(pubkeys []string) []bls_cosi.PublicKeyWrapper {
+	blsPubkeys := make([]bls_cosi.PublicKeyWrapper, len(pubkeys))
+	for i := range pubkeys {
+		if key, err := bls_cosi.WrapperPublicKeyFromString(pubkeys[i]); err != nil {
+			panic(fmt.Sprintf("invalid bls key: %d:%s error:%s", i, pubkeys[i], err.Error()))
+		} else {
+			blsPubkeys[i] = *key
+		}
+	}
+	return blsPubkeys
+}
+
+// each time to update the allowlist, it requires a hardfork.
+var mainnetAllowlist = Allowlist{
```

Comment:
AllowList should have a clear versioning too just like the sharding config instance. Used allowList can be changed anymore. So put like mainnetAllowlistV1 for example.